internal/provider/models/processors/set_timestamp.go

```diff
@@ -0,0 +1,156 @@
+)
+
+const SET_TIMESTAMP_PROCESSOR_TYPE_NAME = "set_timestamp"
+const SET_TIMESTAMP_PROCESSOR_NODE_NAME = "set-timestamp"
+
+type SetTimestampProcessorModel struct {
+	Id           StringValue `tfsdk:"id"`
+	PipelineId   StringValue `tfsdk:"pipeline_id"`
+	Title        StringValue `tfsdk:"title"`
+	Description  StringValue `tfsdk:"description"`
+	Inputs       ListValue   `tfsdk:"inputs"`
+	GenerationId Int64Value  `tfsdk:"generation_id"`
+	Parsers      ListValue   `tfsdk:"parsers" user_config:"true"`
+}
+
+var parserAttrTypes = map[string]attr.Type{
+	"field":                   StringType{},
+	"timestamp_format":        StringType{},
+	"custom_timestamp_format": StringType{},
+}
+
+var SetTimestampProcessorResourceSchema = schema.Schema{
+	Description: "Parse a list of fields using a chosen time format. First match will override the default timestamp of the event.",
+	Attributes: ExtendBaseAttributes(map[string]schema.Attribute{
+		"parsers": schema.ListNestedAttribute{
+			Required:    true,
+			Description: "The list of fields and parsers to use in order of priority, short-circuiting on the first successful match.",
+			Validators: []validator.List{
+				listvalidator.SizeAtLeast(1),
+			},
+			NestedObject: schema.NestedAttributeObject{
+				Attributes: map[string]schema.Attribute{
+					"field": schema.StringAttribute{
+						Required:    true,
+						Description: "The field whose value will be parsed.",
+						Validators: []validator.String{
+							stringvalidator.LengthAtLeast(1),
+						},
+					},
+					"timestamp_format": schema.StringAttribute{
+						Required:    true,
+						Description: "The timestamp format.",
+						Validators: []validator.String{
+							stringvalidator.LengthAtLeast(1),
+						},
+					},
+					"custom_timestamp_format": schema.StringAttribute{
+						Optional:    true,
+						Description: "Custom timestamp format, according to strftime.",
+						Validators: []validator.String{
+							stringvalidator.LengthAtLeast(1),
+						},
+					},
+				},
+			},
+		},
+	}),
+}
+
+func SetTimestampProcessorFromModel(plan *SetTimestampProcessorModel, previousState *SetTimestampProcessorModel) (*Processor, diag.Diagnostics) {
+	dd := diag.Diagnostics{}
+	component := Processor{
+		BaseNode: BaseNode{
+			Type:        SET_TIMESTAMP_PROCESSOR_NODE_NAME,
+			Title:       plan.Title.ValueString(),
+			Description: plan.Description.ValueString(),
+			UserConfig:  map[string]any{},
+		},
+	}
+
+	if previousState != nil {
+		component.Id = previousState.Id.ValueString()
+		component.GenerationId = previousState.GenerationId.ValueInt64()
+	}
+	component.Inputs = StringListValueToStringSlice(plan.Inputs)
+
+	var parsers []map[string]any
+	for _, v := range plan.Parsers.Elements() {
+		parserItem := make(map[string]any)
+		optionsItem := make(map[string]string)
+
+		parserPlanMap := MapValuesToMapAny(v, &dd)
+		format := parserPlanMap["timestamp_format"].(string)
```

Comment:
issue: if the user provides an epoch as an int this will fail

suggestion: add an error if the timestamp is not a string

```suggestion
		format, ok := parserPlanMap[TimestampFormatKey].(string)
		if !ok {
		...
		}
```